sematic/api/endpoints/storage.py

```diff
@@ -80,7 +84,8 @@ def get_stored_data_redirect(user: Optional[User], namespace: str, key: str):
 
     destination = storage_plugin().get_read_destination(namespace, key, user)
 
-    response = flask.redirect(destination.url, code=HTTPStatus.FOUND)
+    url = _get_storage_destination_url(destination, flask.request)
```

Comment:
Please consider encapsulating this function in `StorageDestination`.
```suggestion
    url = destination.get_storage_destination_url(request.args.get("origin"))
```